### PR TITLE
Add support for youch addLink feature

### PR DESCRIPTION
**Pull request by emanuel-lundman**
Don't know if this is something you're interested in, or (if you are) if you want it implemented in a different way. But here's a quick solution using youch's addLink and an optional options object for the errorReporter with an addStackOverflowLink boolean option.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -30,6 +30,9 @@ type IReceivedError = Partial<Error> & string & {
 };
 
 type ResponseFormat = 'json' | 'text' | 'html';
+interface ErrorReporterOptions {
+    addStackOverflowLink?: boolean;
```
Comment (hmil) [inline src/index.ts]: Remove the question mark. Then in the exported function `errorReporter`, use `Partial<ErrorReporterOptions>`.

File: src/index.ts [hunk 2]
```diff
@@ -42,13 +45,18 @@ type ResponseFormat = 'json' | 'text' | 'html';
    app.use(errorReporter());
    ```
  */
-export function errorReporter() {
+export function errorReporter(options?: ErrorReporterOptions) {
```
Comment (hmil) [inline src/index.ts]: Create a local `const realOptions: ErrorReporterOptions = {...defaultOptions, ...options}` (of course, you need to define the default options somewhere).

This allows you to use non-optional types everywhere else in the internal code.

File: src/index.ts [hunk 3]
```diff
@@ -113,14 +126,22 @@ function getPreferredResponseFormat(req: Request): ResponseFormat {
...
+  options?: ErrorReporterOptions
+): Promise<void> {
     const youch = new Youch(err, req);
-    return youch.toHTML()
-        .then((html: string) => {
-            res.writeHead(err.statusCode, {'content-type': 'text/html'});
-            res.write(html);
-            res.end();
-        });
+    const asHtml =
```
Comment (hmil) [inline src/index.ts]: Here, you could let `youch` be a non-constant variable (eg. `let youchBuilder = new Youch(...`). Then loop over (or reduce) `options.links` to  `addLink` on `youchBuilder` (I am assuming addLink is meant to be called multiple times to add multiple links).

**Comment by hmil** [comment]
Hey! Thanks for the contribution.

I wasn't aware of the `addLink` API, and I think it is a good idea to expose it. However, I don't want to add opinionated code for stackOverflow because this module is meant to be a straightforward integration of youch into express.

Could you replace the `addStackOverflowLink` option with a `links` option of type `Array<({ message: string }) => string>`. This provides consumers of `express-youch` the same capabilities `youch` offers out of the box.

File: src/index.ts (hunk 2 above)
Comment (hmil) [reply src/index.ts]: note: you might have to use `Object.assign` if TypeScript complains about the spread operator.

File: src/index.ts (hunk 3 above)
Comment (emanuel-lundman) [reply src/index.ts]: Fixed. Added a reduce, seemed more readable.

File: src/index.ts (hunk 2 above)
Comment (emanuel-lundman) [reply src/index.ts]: fixed

File: src/index.ts (hunk 1 above)
Comment (emanuel-lundman) [reply src/index.ts]: fixed

**Comment by emanuel-lundman** [comment]
Hi. Tweaked code based on your requests and tested it locally using:
```typescript
.use(errorReporter({ links: [
    ({message}) => `<a href="#test1">1 ${message}</a>`,
    ({message}) => `<a href="#test2">2 ${message}</a>`,
    ({message}) => `<a href="#test3">3 ${message}</a>`,
    ({message}) => `<a href="#test4">4 ${message}</a>`,
]}));
```
Seemed to work fine.
Feel free to change anything to your liking.

**Comment by hmil** [comment]
Thanks for the high quality submission 💯  . I appreciate it.

**Comment by hmil** [comment]
ps: consider starring the repo if you like this package ;)

**Pull request merged**